src/types/formiz-core/index.d.ts

```diff
@@ -0,0 +1,46 @@
+declare module '@formiz/core' {
+  export * from '@formiz/core/dist';
+
+  import {
+    UseFieldProps,
+    UseFieldValues,
+  } from '@formiz/core/dist/types/field.types';
+
+  // Override FieldProps type with generic param for define value type
+  export declare type FieldProps<Value = ExplicitAny> = Omit<
+    UseFieldProps,
+    'defaultValue' | 'onChange' | 'formatValue'
+  > & {
+    defaultValue?: Value;
+    onChange?: (value: Value, rawValue?: Value) => void;
+    formatValue?: (value: Value) => unknown;
+  };
+
+  // Override useField type for value and otherProps type inferred by props
+  export declare const useField: <
+    Props extends FieldProps = FieldProps,
+    Value = Required<Props>['defaultValue'] | null
+  >(
+    props: Props
+  ) => Omit<UseFieldValues, 'otherProps' | 'value' | 'setValue'> & {
+    otherProps: Omit<Props, keyof FieldProps>;
+    value: Value;
+    setValue(value: Value | ((prevValue: Value) => Value)): void;
```

Comment:
vu que c'est la meme systeme de state on peut pas faire ça ?
```suggestion
    Dispatch<SetStateAction< Value>>
```

Reply:
It's done
```
React.Dispatch<React.SetStateAction<Value>>
```
Thanks !